pngencoder.lua

```diff
@@ -121,11 +124,20 @@ function Png:adler32(data, index, len)
     self.adler = bit.bor(bit.lshift(s2, 16), s1)
 end
 
-local function begin(width, height)
-    local state = setmetatable({ width = width, height = height, done = false, output = {} }, Png)
+local function begin(width, height, colorType)
```

Comment:
If the global enums are removed, the `colorType` could be eg. a boolean `hasAlpha` or a string accepting either "rgb" or "rgba".